src/platform_linux/shmem.c

```diff
@@ -585,7 +666,38 @@ platform_shm_free(platform_heap_id hid,
       return;
    }
 
-   platform_shm_track_free(shm, ptr, objname, func, file, lineno);
+   // Micro-optimization for very-last-fragment-allocated being freed
+   bool   maybe_large_frag = TRUE;
+   size_t frag_size        = 0;
+
+   shm_lock_mem(shm);
+   shm->shm_nfrees++;
+   if (shm->shm_last_alloc == ptr) {
```

Comment:
If small fragment being freed was the very last one allocated, re-allocate it by grabbing that fragment into 'free space'.

This single optimization pays off hugely for `build/release/bin/unit/splinter_test --use-shmem --num-inserts 20000000 test_lookups`

```
Deallocated SplinterDB shared memory segment at 0x7f3996e51000, shmid=0. Used=3291704704 bytes (~3.06 GiB, 76.64 %) Used HWM=3291704768 bytes (~3.06 GiB, 76.64 %), Free=1003256256 bytes (~956.77 MiB, 23.36 %).
SplinterDB shared memory stats: nfrees=20000706, nfrees-last-small-frag=20000101 (100.00 %), Large fragments in-use HWM=11 (found in-use=0), consumed=3291363008 bytes (~3.06 GiB).
```

Almost all the frees were for the very last fragment allocated; no concurrency in the workload.